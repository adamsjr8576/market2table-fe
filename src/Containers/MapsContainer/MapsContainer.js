import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Route, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './MapsContainer.scss';
import { Map, GoogleApiWrapper } from 'google-maps-react';
import { InfoWindow, Marker } from 'google-maps-react';

const mapStyles = {
  width: '900px',
  height: '600px'
};

let market;

export const MapsContainer = ({ google }) => {

  const [selectedPlace, setSelectedPlace] = useState({});
  const [activeMarker, setActiveMarker] = useState({});
  const [showingInfoWindow, setShowingInfoWindow] = useState(false);
  const markets = useSelector(state => state.markets);
  const selectedMarketId = useSelector(state => state.selectedMarket);

  const onMarkerClick = (props, marker, e) => {
    setSelectedPlace(props);
    setActiveMarker(marker);
    setShowingInfoWindow(true);
  }

  const onClose = props => {
   if (showingInfoWindow) {
     setShowingInfoWindow(false);
     setActiveMarker(null);
   }
  };

  const setMarkers = () => {
    if (selectedMarketId.length === 7) {
      const marketInfo = markets.find(market => market.id === selectedMarketId);
      return (<Marker
          onClick={onMarkerClick}
          name={marketInfo.marketname}
          position={{ lat: marketInfo.latitude, lng: marketInfo.longitude }}
          key={marketInfo.id}
        />)
    } else {
      return markets.map(market => {
        return (
          <Marker
              onClick={onMarkerClick}
              name={market.marketname}
              position={{ lat: market.latitude, lng: market.longitude }}
              key={market.id}
            />
        )
      })
    }
  }

  const setCenter = () => {
    if (selectedMarketId.length === 7) {
      const marketInfo = markets.find(market => market.id === selectedMarketId);
      return {
        lat: marketInfo.latitude,
        lng: marketInfo.longitude
      }
    } else {
      return {
       lat: markets[0].latitude,
       lng: markets[0].longitude
      }
    }
  }

  const setZoom = () => {
    if (selectedMarketId.length === 7) {
      return 14
    } else {
      return 10
    }
  }

  let toReturn;
  if (markets.length) {
    const center = setCenter();
    const marketPins = setMarkers();
    const zoom = setZoom();
    toReturn = (
      <div id='mapBox' className='map-container'>
        <Map
          google={google}
          zoom={zoom}
          style={mapStyles}
          initialCenter={center}
        >
          {marketPins}
          <InfoWindow
            marker={activeMarker}
            visible={showingInfoWindow}
            onClose={onClose}
          >
            <div className='info-window-container'>
              <h4>{selectedPlace.name}</h4>
            </div>
          </InfoWindow>
        </Map>
      </div>)
  } else {
    toReturn = (<p>LOADING</p>);
  }

  return toReturn
}


export default GoogleApiWrapper({
  apiKey: process.env.REACT_APP_API_KEY
})(MapsContainer);
