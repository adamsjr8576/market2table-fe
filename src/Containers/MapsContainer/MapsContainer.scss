#mapBox div:first-child {
    position: relative !important;
}

.map-container {
  width: 75%;
  height: 100%;
}

.info-window-container {
  display: flex;
  align-items: center;
  justify-content: space-around;
}
