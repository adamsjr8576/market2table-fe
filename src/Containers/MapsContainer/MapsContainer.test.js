import React from 'react';
import { shallow } from 'enzyme';
import { MapsContainer } from './MapsContainer';
import { useSelector, useDispatch } from 'react-redux';

let mockState = {
  zipCode: '80401',
  selectedMarket: '1923782',
  markets: [{name: 'Friendly Farms Market', id:'1923782'}]
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState
}));

describe('MapsContainer', () => {

  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallow(<MapsContainer />);

    expect(wrapper).toMatchSnapshot();
  });
});
