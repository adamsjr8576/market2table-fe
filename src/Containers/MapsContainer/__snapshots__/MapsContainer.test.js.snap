// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapsContainer should match the snapshot 1`] = `
<div
  className="map-container"
  id="mapBox"
>
  <Map
    center={
      Object {
        "lat": undefined,
        "lng": undefined,
      }
    }
    centerAroundCurrentLocation={false}
    containerStyle={Object {}}
    initialCenter={
      Object {
        "lat": undefined,
        "lng": undefined,
      }
    }
    style={
      Object {
        "height": "600px",
        "width": "900px",
      }
    }
    visible={true}
    zoom={14}
  >
    <Marker
      key="1923782"
      name="Marker"
      onClick={[Function]}
      position={
        Object {
          "lat": undefined,
          "lng": undefined,
        }
      }
    />
    <InfoWindow
      marker={Object {}}
      onClose={[Function]}
      visible={false}
    >
      <div
        className="info-window-container"
      >
        <h4 />
      </div>
    </InfoWindow>
  </Map>
</div>
`;
